app/all-listings/page.tsx

```
@@ -65,104 +65,167 @@ function AllListingsContent() {
     } catch (error) {
-      toast.error("Failed to fetch properties");
+      console.error("Failed to fetch cities:", error);
+      toast.error("Failed to load cities");
     } finally {
-      setLoading(false);
+      setCitiesLoading(false);
     }
+  }, []);
+
+  // Debounced search function
+  const debounce = (func: Function, wait: number) => {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)